src/hera.cpp

```diff
@@ -341,6 +341,8 @@ evm_result evm_execute(
     ret.gas_left = result.gasLeft;
   } catch (OutOfGasException const&) {
     ret.status_code = EVM_OUT_OF_GAS;
+  } catch (StaticModeViolation const&) {
+    ret.status_code = EVM_STATIC_MODE_ERROR;
```

Comment:
Actually, since you added the reason, it makes sense to put a `HERA_DEBUGGING` case here displaying the reason.